### PR TITLE
docker-compose files to create a 4 validating peer consensus network

### PR DESCRIPTION
**Pull request by tuand27613**
signed by Tuan tdang@us.ibm.com

### CODE REVIEW

**Comment by tuand27613** [comment]
Signed-off: Tuan Dang tdang@us.ibm.com

File: common/metrics/gendoc/options.go [hunk 1]
```diff
@@ -167,7 +167,7 @@ func populateOption(lit *ast.CompositeLit, target interface{}) (interface{}, err
 				field.Set(labelNamesValue)
 
 			// simple strings
-			case "Namespace", "Subsystem", "Name", "Help", "StatsdFormat":
+			case "Namespace", "Subsystem", "Name", "Help", "StatsdFormat", "SourceComponent":
```
Comment (sykesm) [inline common/metrics/gendoc/options.go]: Names are always difficult, but `SourceComponent` does not reflect the intent. It's also confusing to see `parentComponent` and `SourceComponent` used as synonyms within the same pull request.

The request was for us to identify whether a metric is emitted by the peer, the orderer, or both (common). Using a field on the metric is fine but it also seems like the namespace would imply the grouping. (I'll come back to this in a bit.)

File: common/metrics/gendoc/table.go [hunk 2]
```diff
@@ -21,11 +21,12 @@ import (
...
-	Type                     // Type is the type of meter option.
-	Description              // Description is the help text from the meter option.
-	Labels                   // Labels is the meter's label information.
-	Bucket                   // Bucket is the statsd bucket format
+	Name            Field = iota // Name is the meter name.
+	Type                         // Type is the type of meter option.
+	Description                  // Description is the help text from the meter option.
+	Labels                       // Labels is the meter's label information.
+	Bucket                       // Bucket is the statsd bucket format
+	SourceComponent              // peer or orderer
```
Comment (sykesm) [inline common/metrics/gendoc/table.go]: This isn't exclusive. Some metrics (eg. comm layer, logging counts) are common to both; others may also be present in the CA.

File: common/metrics/gendoc/table.go [hunk 3]
```diff
@@ -51,6 +52,7 @@ func NewPrometheusTable(cells Cells) Table {
 			{Field: Name, Name: "Name", Width: max(cells.MaxLen(Name)+2, 20)},
 			{Field: Type, Name: "Type", Width: 11},
 			{Field: Description, Name: "Description", Width: 60},
+			{Field: SourceComponent, Name: "Parent Component", Width: 20},
```
Comment (sykesm) [inline common/metrics/gendoc/table.go]: Here's where we use `SourceComponent` and "Parent Component" as synonyms. Between the inconsistency and the possible inaccuracy, I wonder if there's a better breakdown?

**Pull request merged**